From 4296c8c32939484948d7461c6c474e6319cfe884 Mon Sep 17 00:00:00 2001
From: Alexandru Costache <alexandru@balena.io>
Date: Wed, 17 Aug 2022 16:08:54 +0200
Subject: [PATCH] cw2015: fix build for nrt

The cw2015 driver is backported from a newer kernel,
we need to use dev_err because dev_err_probe is not
available in 5.4.

Upstream-status: Inappropriate [configuration]
Signed-off-by: Alexandru Costache <alexandru@balena.io>
---
 drivers/power/supply/cw2015_battery.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/power/supply/cw2015_battery.c b/drivers/power/supply/cw2015_battery.c
index d5231b49e1c1..53ddef2c1a5c 100644
--- a/drivers/power/supply/cw2015_battery.c
+++ b/drivers/power/supply/cw2015_battery.c
@@ -674,8 +674,7 @@ static int cw_bat_probe(struct i2c_client *client)
 						    &psy_cfg);
 	if (IS_ERR(cw_bat->rk_bat)) {
 		/* try again if this happens */
-		dev_err_probe(&client->dev, PTR_ERR(cw_bat->rk_bat),
-			"Failed to register power supply\n");
+		dev_err(&client->dev, "Failed to register power supply\n");
 		return PTR_ERR(cw_bat->rk_bat);
 	}
 
-- 
2.37.2

